src/rng.rs

```diff
@@ -9,20 +9,30 @@ use rand_core::{RngCore, SeedableRng};
 /// and the new seed material.
 // TODO: later: re-evaluate decision about ChaChaRng
 pub trait FiatShamirRng: RngCore {
+    /// Internal representation of the seed, useful to get and set
+    type Seed: Clone;
+
     /// initialize the RNG
     fn new() -> Self;
 
-    /// Create a new `Self` by initializing with a fresh seed.
+    /// Create a new `Self` by initializing its internal state with a fresh `seed`,
+    /// generically being something serializable to a byte array.
     fn from_seed<'a, T: 'a + ToBytes>(seed: &'a T) -> Self;
 
-    /// Refresh `self.seed` with new material. Achieved by setting
-    /// `self.seed = H(self.seed || new_seed)`.
+    /// Refresh the internal state with new material `seed`, generically being
+    /// something serializable to a byte array.
     fn absorb<'a, T: 'a + ToBytes>(&mut self, seed: &'a T);
 
     /// Squeeze a new random field element
     fn squeeze_128_bits_challenge<F: Field>(&mut self) -> F {
         u128::rand(self).into()
     }
+
+    /// Get the internal state in the form of an instance of `Self::Seed`.
+    fn get_seed(&self) -> &Self::Seed;
+
+    /// Set interal state according to the specified `new_seed`
+    fn set_seed(&mut self, new_seed: Self::Seed);
```

Comment:
Why don't we call it ``get_state`` and ``set_state``?
What are the functions intended for?

Reply:
I use them only in the batch check batch proofs as a way to hand over the state. Maybe it's a good proposal to call it get state and set state you're right